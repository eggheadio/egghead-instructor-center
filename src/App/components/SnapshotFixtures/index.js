import React from 'react'

export const StringFixture = 'String'

export const ImageUrlFixture = 'https://s3.amazonaws.com/uifaces/faces/twitter/kimcool/128.jpg'

export const ComponentFixture = () => (
  <div>
    Component
  </div>
)
