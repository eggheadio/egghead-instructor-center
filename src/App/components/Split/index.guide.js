import React from 'react'
import {storiesOf} from '@kadira/storybook'
import {lorem} from 'faker'
import Split from '.'

storiesOf('Split', module)
  .add('default', () => (
    <Split
      title={lorem.words()}
      main={lorem.text()}
      aside={lorem.text()}
    />
  ))
