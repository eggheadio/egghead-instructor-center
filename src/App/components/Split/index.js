import React, {PropTypes} from 'react'
import Heading from '../Heading'

const Split = ({
  title,
  main,
  aside,
}) => (
  <div>

    <Heading level='1'>
      {title}
    </Heading>

    <div className='flex-ns'>

      <main className='mr4-ns w-50-ns'>
        {main}
      </main>

      <aside className='w-50-ns mt5 mt0-ns'>
        {aside}
      </aside>

    </div>

  </div>
)

Split.propTypes = {
  title: PropTypes.node.isRequired,
  main: PropTypes.node.isRequired,
  aside: PropTypes.node.isRequired,
}

export default Split
