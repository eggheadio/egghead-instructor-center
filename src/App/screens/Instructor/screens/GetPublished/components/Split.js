import React from 'react'
import Title from './Title'

const Split = ({
  title,
  main,
  aside,
}) => (
  <div>

    <Title>
      {title}
    </Title>

    <div className='flex-ns'>

      <main
        className='mr4-ns'
        style={{flex: 1}}
      >
        {main}
      </main>

      <aside style={{flex: 1}}>
        {aside}
      </aside>

    </div>

  </div>
)

Split.propTypes = {
  main: React.PropTypes.node.isRequired,
  aside: React.PropTypes.node.isRequired,
}

export default Split
