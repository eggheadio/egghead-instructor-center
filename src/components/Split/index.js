// TODO: Consume from egghead-ui once published

import React, {PropTypes} from 'react'
import Heading from 'components/Heading'

const Split = ({
  intro,
  title,
  main,
  aside,
}) => (
  <div>

    {intro}

    <Heading level='1'>
      {title}
    </Heading>

    <div className='flex-ns'>

      <main className='mr4-ns w-50-ns'>
        {main}
      </main>

      <aside className='w-50-ns mt5 mt0-ns'>
        {aside}
      </aside>

    </div>

  </div>
)

Split.propTypes = {
  intro: PropTypes.node,
  title: PropTypes.string.isRequired,
  main: PropTypes.node.isRequired,
  aside: PropTypes.node.isRequired,
}

export default Split
