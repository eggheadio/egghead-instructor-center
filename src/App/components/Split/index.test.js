import React from 'react'
import renderer from 'react-test-renderer'
import {StringFixture, ComponentFixture} from '../SnapshotFixtures'
import Split from '.'

test('Split', () => {
  const tree = renderer.create(
    <Split
      title={StringFixture}
      main={<ComponentFixture />}
      aside={<ComponentFixture />}
    />
  )
  expect(tree).toMatchSnapshot()
})
