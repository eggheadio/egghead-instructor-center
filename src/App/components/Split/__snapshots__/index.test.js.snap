exports[`test default 1`] = `
<div>
  <div>
    <h1
      className="mt0 normal f1 mb4">
      String
    </h1>
  </div>
  <div
    className="flex-ns">
    <main
      className="mr4-ns w-50-ns">
      <div>
        Component
      </div>
    </main>
    <aside
      className="w-50-ns mt5 mt0-ns">
      <div>
        Component
      </div>
    </aside>
  </div>
</div>
`;
